pkg/planner/core/operator/physicalop/physical_indexlookup.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package physicalop
+
+import (
+	"fmt"
+	"unsafe"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/pkg/kv"
+	"github.com/pingcap/tidb/pkg/planner/core/base"
+	"github.com/pingcap/tidb/pkg/util/plancodec"
+	"github.com/pingcap/tidb/pkg/util/size"
+	"github.com/pingcap/tipb/go-tipb"
+)
+
+// PhysicalIndexLookUp represents a index lookup
+// It contains two parts: indexPlan and tablePlan
+// IndexPlan maybe hierarchical, e.g. IndexScan -> Selection -> Limit
+// TablePlan contains only TableScan executor
+type PhysicalIndexLookUp struct {
+	PhysicalSchemaProducer
+	// handle offsets in the indexPlan's output schema
+	IndexHandleOffsets []uint32
+}
+
+func resetPlanIDRecursively(ctx base.PlanContext, p base.PhysicalPlan) {
+	p.SetID(int(ctx.GetSessionVars().PlanID.Add(1)))
+	for _, child := range p.Children() {
+		resetPlanIDRecursively(ctx, child)
+	}
+}
+
+func buildPushDownIndexLookUpPlan(
+	ctx base.PlanContext, indexPlan base.PhysicalPlan, tablePlan base.PhysicalPlan,
+) (indexLookUpPlan base.PhysicalPlan, err error) {
+	tablePlan, err = tablePlan.Clone(ctx)
+	if err != nil {
+		return nil, err
+	}
+	resetPlanIDRecursively(ctx, tablePlan)
+
+	tableScanPlan, parentOfTableScan, err := detachRootTableScanPlan(tablePlan)
+	if err != nil {
+		return nil, err
+	}
+
+	// to make sure tableScanPlan can be cloned,
+	// runtimeFilterList has tag `plan-cache-clone:"must-nil"`
+	if len(tableScanPlan.runtimeFilterList) == 0 {
+		tableScanPlan.runtimeFilterList = nil
+	}
+
+	// to make sure tableScanPlan can be cloned,
+	// UsedColumnarIndexes has tag `plan-cache-clone:"must-nil"`
+	if len(tableScanPlan.UsedColumnarIndexes) == 0 {
+		tableScanPlan.UsedColumnarIndexes = nil
+	}
```

Comment:
These are attributes of tiflash scan(or the future tici scan).
And actually, you will see that the Clone method will throw FALSE when it's a tiflash scan.
So you can remove the check.